sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -198,6 +203,66 @@ case class StringSplit(str: Expression, pattern: Expression)
   override def prettyName: String = "split"
 }
 
+/**
+ * Splits a string into arrays of sentences, where each sentence is an array of words.
+ * The 'lang' and 'country' arguments are optional, and if omitted, the default locale is used.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(str, lang, country) - Splits str into an array of array of words.",
+  extended = "> SELECT _FUNC_('Hi there! Good morning.');\n  [['Hi','there'], ['Good','morning']]")
+case class Sentences(
+    str: Expression,
+    language: Expression = Literal(""),
+    country: Expression = Literal(""))
+  extends Expression with ImplicitCastInputTypes with CodegenFallback {
+
+  def this(str: Expression) = this(str, Literal(""), Literal(""))
+  def this(str: Expression, language: Expression) = this(str, language, Literal(""))
+
+  override def nullable: Boolean = true
+  override def dataType: DataType =
+    ArrayType(ArrayType(StringType, containsNull = false), containsNull = false)
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType, StringType)
+  override def children: Seq[Expression] = str :: language :: country :: Nil
+
+  override def eval(input: InternalRow): Any = {
+    val string = str.eval(input)
+    if (string == null) {
+      null
+    } else {
+      val locale = try {
+        new Locale(language.eval(input).asInstanceOf[UTF8String].toString,
+          country.eval(input).asInstanceOf[UTF8String].toString)
+      } catch {
+        case _: NullPointerException | _: ClassCastException => Locale.getDefault
```

Comment:
It created the wrong locale, then it compared the system available locales. So, finally, ignored. The following is the underlying code in `rt.jar`.
```
            List var4 = Control.getControl(Control.FORMAT_DEFAULT).getCandidateLocales("", var1);
            Iterator var5 = var4.iterator();

            while(var5.hasNext()) {
                Locale var6 = (Locale)var5.next();
                if(!var6.equals(var1)) {
                    var2 = findAdapter(var0, var6);
                    if(var2 != null) {
                        ((ConcurrentMap)var3).putIfAbsent(var1, var2);
                        return var2;
                    }
                }
            }
```